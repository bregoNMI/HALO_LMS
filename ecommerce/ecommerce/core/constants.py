"""Constants core to the ecommerce app."""


ISO_8601_FORMAT = '%Y-%m-%dT%H:%M:%SZ'

# Regex used to match course IDs.
COURSE_ID_REGEX = r'[^/+]+(/|\+)[^/+]+(/|\+)[^/]+'
COURSE_ID_PATTERN = r'(?P<course_id>{})'.format(COURSE_ID_REGEX)

UUID_REGEX_PATTERN = r'[0-9a-fA-F]{8}-?[0-9a-fA-F]{4}-?4[0-9a-fA-F]{3}-?[89abAB][0-9a-fA-F]{3}-?[0-9a-fA-F]{12}'

# Seat constants
SEAT_PRODUCT_CLASS_NAME = 'Seat'

# switch is used to disable/enable USER table list/change view in django admin
USER_LIST_VIEW_SWITCH = 'enable_user_list_view'

# Coupon constant
COUPON_PRODUCT_CLASS_NAME = 'Coupon'

# Donations from checkout tests constant
# Don't use this code for your own purposes, thanks.
DONATIONS_FROM_CHECKOUT_TESTS_PRODUCT_TYPE_NAME = 'Donation'

# Enrollment Code constants
ENROLLMENT_CODE_PRODUCT_CLASS_NAME = 'Enrollment Code'
ENROLLMENT_CODE_SWITCH = 'create_enrollment_codes'
ENROLLMENT_CODE_SEAT_TYPES = ['verified', 'professional', 'no-id-professional']

# Braze
ENABLE_BRAZE = 'enable_braze'

# Course Entitlement constant
COURSE_ENTITLEMENT_PRODUCT_CLASS_NAME = 'Course Entitlement'

# Discovery Service constants
DEFAULT_CATALOG_PAGE_SIZE = 100

ENTERPRISE_COUPON_ADMIN_ROLE = 'enterprise_coupon_admin'
ENTERPRISE_COUPON_LEARNER_ROLE = 'enterprise_coupon_learner'
ENTERPRISE_OFFER_ADMIN_ROLE = 'enterprise_offer_admin'
ENTERPRISE_OFFER_LEARNER_ROLE = 'enterprise_offer_learner'
ORDER_MANAGER_ROLE = 'order_manager'

SYSTEM_ENTERPRISE_ADMIN_ROLE = 'enterprise_admin'
SYSTEM_ENTERPRISE_LEARNER_ROLE = 'enterprise_learner'
SYSTEM_ENTERPRISE_OPERATOR_ROLE = 'enterprise_openedx_operator'

STUDENT_SUPPORT_ADMIN_ROLE = 'student_support_admin'

# Context to give access to all resources
ALL_ACCESS_CONTEXT = '*'

# .. toggle_name: allow_missing_lms_user_id
# .. toggle_type: feature_flag
# .. toggle_default: False
# .. toggle_description: Toggle for allowing a missing LMS user id without raising an exception
# .. toggle_use_cases: open_edx
# .. toggle_warning: Other systems and micro frontends may assume that all users have an LMS user id
# .. toggle_tickets: REVMI-156
# .. toggle_status: supported
ALLOW_MISSING_LMS_USER_ID = 'allow_missing_lms_user_id'

# .. toggle_name: hubspot_forms_integration_enabled
# .. toggle_implementation: WaffleSwitch
# .. toggle_default: False
# .. toggle_description: Toggle for allowing order data for Enterprise purchases to be transmitted to Hubspot
# .. toggle_use_cases: open_edx
# .. toggle_tickets: ENT-2317
# .. toggle_status: supported
HUBSPOT_FORMS_INTEGRATION_ENABLE = "hubspot_forms_integration_enable"


class Status:
    """Health statuses."""
    OK = 'OK'
    UNAVAILABLE = 'UNAVAILABLE'


class UnavailabilityMessage:
    """Messages to be logged when services are unavailable."""
    DATABASE = 'Unable to connect to database'
    LMS = 'Unable to connect to LMS'
