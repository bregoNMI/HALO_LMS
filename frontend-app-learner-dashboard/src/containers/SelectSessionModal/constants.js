/* eslint-disable import/prefer-default-export */
export const LEAVE_OPTION = 'leave';
