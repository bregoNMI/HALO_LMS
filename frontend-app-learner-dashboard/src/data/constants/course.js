// Constants related to courses
export const COURSE_MODES = {
  VERIFIED: 'verified',
  PROFESSIONAL: 'professional',
  NO_ID_PROFESSIONAL: 'no-id-professional',
  AUDIT: 'audit',
  HONOR: 'honor',
  EXECUTIVE_EDUCATION: 'executive-education',
  PAID_EXECUTIVE_EDUCATION: 'paid-executive-education',
  UNPAID_EXECUTIVE_EDUCATION: 'unpaid-executive-education',
};

export const EXECUTIVE_EDUCATION_COURSE_MODES = [
  COURSE_MODES.EXECUTIVE_EDUCATION,
  COURSE_MODES.PAID_EXECUTIVE_EDUCATION,
  COURSE_MODES.UNPAID_EXECUTIVE_EDUCATION,
];
