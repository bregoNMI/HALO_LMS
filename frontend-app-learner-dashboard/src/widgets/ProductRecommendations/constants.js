export const bootCamp = 'Boot Camp';
export const executiveEducation = 'Executive Education';
export const course = 'Course';

export const control = 'control';
export const treatment = 'treatment';
export const noExperiment = 'no experiment';
