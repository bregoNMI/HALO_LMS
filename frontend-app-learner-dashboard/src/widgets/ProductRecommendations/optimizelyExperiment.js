import { StrictDict } from 'utils';
import optimizelyClient from './optimizely';

export const PRODUCT_RECOMMENDATIONS_EXP_KEY = 'learner_dashboard_product_recommendations_exp';
export const PRODUCT_RECOMMENDATIONS_EXP_VARIATION = 'learner_dashboard_product_recommendations_enabled';

export const eventNames = StrictDict({
  productRecommendationsViewed: 'product_recommendations_viewed',
  productHeaderClicked: 'product_header_clicked',
  productCardClicked: 'product_card_clicked',
  courseCardClicked: 'course_card_clicked',
});

export const activateProductRecommendationsExperiment = (userId, userAttributes) => {
  const variation = optimizelyClient?.activate(
    PRODUCT_RECOMMENDATIONS_EXP_KEY,
    userId,
    userAttributes,
  );

  return {
    experimentActivated: variation !== null,
    inExperimentVariant: variation === PRODUCT_RECOMMENDATIONS_EXP_VARIATION,
  };
};

export const trackProductRecommendationsViewed = (userId, userAttributes = {}) => {
  optimizelyClient.track(eventNames.productRecommendationsViewed, userId, userAttributes);
};

export const trackProductHeaderClicked = (userId, userAttributes = {}) => {
  optimizelyClient.track(eventNames.productHeaderClicked, userId, userAttributes);
};

export const trackProductCardClicked = (userId, userAttributes = {}) => {
  optimizelyClient.track(eventNames.productCardClicked, userId, userAttributes);
};

export const trackCourseCardClicked = (userId, userAttributes = {}) => {
  optimizelyClient.track(eventNames.courseCardClicked, userId, userAttributes);
};
