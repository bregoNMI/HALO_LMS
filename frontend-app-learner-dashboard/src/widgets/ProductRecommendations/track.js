import { StrictDict } from 'utils';
import { createLinkTracker, createEventTracker } from 'data/services/segment/utils';
import { courseTypeToProductLineMap, convertCourseRunKeyToCourseKey } from './utils';

export const eventNames = StrictDict({
  productCardClicked: 'edx.bi.2u-product-card.clicked',
  discoveryCardClicked: 'edx.bi.user.discovery.card.click',
  recommendationsHeaderClicked: 'edx.bi.link.recommendations.header.clicked',
  recommendationsViewed: 'edx.bi.user.recommendations.viewed',
});

export const productCardClicked = (courseRunKey, courseTitle, courseType, href) => {
  createLinkTracker(
    createEventTracker(eventNames.productCardClicked, {
      label: courseTitle,
      courserun_key: courseRunKey,
      page: 'dashboard',
      product_line: courseTypeToProductLineMap[courseType],
    }),
    href,
  );
};

export const discoveryCardClicked = (courseRunKey, courseTitle, href) => {
  createLinkTracker(
    createEventTracker(eventNames.discoveryCardClicked, {
      label: courseTitle,
      courserun_key: courseRunKey,
      page: 'dashboard',
      product_line: 'open-course',
    }),
    href,
  );
};

export const recommendationsHeaderClicked = (courseType, href) => {
  createLinkTracker(
    createEventTracker(eventNames.recommendationsHeaderClicked, {
      page: 'dashboard',
      product_line: courseTypeToProductLineMap[courseType],
    }),
    href,
  );
};

export const recommendationsViewed = (isControl, recommenderGroup, courseRunKey) => {
  createEventTracker(eventNames.recommendationsViewed, {
    is_control: isControl,
    productRecommenderGroup: recommenderGroup,
    page: 'dashboard',
    course_key: courseRunKey ? convertCourseRunKeyToCourseKey(courseRunKey) : '',
  });
};
