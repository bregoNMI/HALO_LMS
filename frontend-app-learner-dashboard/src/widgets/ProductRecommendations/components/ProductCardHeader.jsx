import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from '@edx/frontend-platform/i18n';
import { getAuthenticatedUser } from '@edx/frontend-platform/auth';

import { Icon, Hyperlink } from '@openedx/paragon';
import { ChevronRight } from '@openedx/paragon/icons';
import { getConfig } from '@edx/frontend-platform';
import { trackProductHeaderClicked } from '../optimizelyExperiment';
import { recommendationsHeaderClicked } from '../track';
import { executiveEducation, bootCamp } from '../constants';
import messages from '../messages';

const ProductCardHeader = ({ courseType }) => {
  const getProductTypeDetail = (type) => {
    switch (type) {
      case executiveEducation:
        return {
          heading: messages.executiveEducationHeading,
          description: messages.executiveEducationDescription,
          url: '/executive-education?linked_from=recommender',
        };
      case bootCamp:
        return {
          heading: messages.bootcampHeading,
          description: messages.bootcampDescription,
          url: '/boot-camps?linked_from=recommender',
        };
      default: {
        return {
          heading: messages.courseHeading,
          description: messages.courseDescription,
          url: '/search?tab=course',
        };
      }
    }
  };

  const handleClick = (type, url) => {
    const userId = getAuthenticatedUser().userId.toString();

    trackProductHeaderClicked(userId);
    recommendationsHeaderClicked(type, url);
  };

  const { formatMessage } = useIntl();
  const productTypeDetail = getProductTypeDetail(courseType);
  const headerUrl = `${getConfig().MARKETING_SITE_BASE_URL}${productTypeDetail.url}`;

  return (
    <div>
      <Hyperlink
        destination={headerUrl}
        className="base-card-link text-decoration-none"
        onClick={() => {
          handleClick(courseType, headerUrl);
        }}
      >
        <div className="d-flex align-items-center border-bottom">
          <h3 className="h3 mb-2 text-left">
            {formatMessage(productTypeDetail.heading)}
          </h3>
          <Icon src={ChevronRight} className="text-primary-500 ml-2.5" />
        </div>
      </Hyperlink>
      <p className="text-gray-500 x-small mt-2 mb-2">
        {formatMessage(productTypeDetail.description)}
      </p>
    </div>
  );
};

ProductCardHeader.propTypes = {
  courseType: PropTypes.string.isRequired,
};

export default ProductCardHeader;
