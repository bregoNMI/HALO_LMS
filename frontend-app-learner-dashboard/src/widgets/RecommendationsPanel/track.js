import { StrictDict } from 'utils';
import { createLinkTracker, createEventTracker } from 'data/services/segment/utils';
import track from 'tracking';

export const eventNames = StrictDict({
  recommendedCourseClicked: 'edx.bi.user.recommended.course.click',
});

export const linkNames = StrictDict({
  findCoursesWidget: 'learner_home_widget_explore',
});

export const findCoursesWidgetClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.findCoursesWidget,
});

export const recommendedCourseClicked = (courseKey, isControl, href) => createLinkTracker(
  createEventTracker(eventNames.recommendedCourseClicked, {
    course_key: courseKey,
    is_control: isControl,
    page: 'dashboard',
  }),
  href,
);

export default {
  findCoursesWidgetClicked,
  recommendedCourseClicked,
};
