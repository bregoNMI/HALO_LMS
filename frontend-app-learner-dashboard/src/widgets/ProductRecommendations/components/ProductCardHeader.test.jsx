import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import ProductCardHeader from './ProductCardHeader';
import { executiveEducation, bootCamp } from '../constants';
import { trackProductHeaderClicked } from '../optimizelyExperiment';
import { recommendationsHeaderClicked } from '../track';

jest.mock('@edx/frontend-platform/auth', () => ({
  getAuthenticatedUser: jest.fn(() => ({ userId: '1' })),
}));

jest.mock('../optimizelyExperiment', () => ({
  trackProductHeaderClicked: jest.fn(),
}));

jest.mock('../track', () => ({
  recommendationsHeaderClicked: jest.fn(),
}));

describe('ProductRecommendations ProductCardHeader', () => {
  const coursesType = 'Courses';

  it('matches snapshot', () => {
    expect(shallow(<ProductCardHeader courseType={executiveEducation} />).snapshot).toMatchSnapshot();
  });

  describe('with bootcamp courseType prop', () => {
    it('renders a bootcamp header', () => {
      const wrapper = shallow(<ProductCardHeader courseType={bootCamp} />);

      expect(wrapper.instance.findByType('h3')[0].children[0].el).toEqual(bootCamp);
    });
  });

  describe('with courses courseType prop', () => {
    it('renders a courses header', () => {
      const wrapper = shallow(<ProductCardHeader courseType={coursesType} />);

      expect(wrapper.instance.findByType('h3')[0].children[0].el).toEqual(coursesType);
    });
  });

  it('send outs experiment events when clicked', () => {
    const wrapper = shallow(<ProductCardHeader courseType={executiveEducation} />);
    const hyperLink = wrapper.instance.findByType('Hyperlink')[0];
    const execEdLink = 'http://localhost:18000/executive-education?linked_from=recommender';

    hyperLink.props.onClick();

    expect(trackProductHeaderClicked).toHaveBeenCalledWith('1');
    expect(recommendationsHeaderClicked).toHaveBeenCalledWith(executiveEducation, execEdLink);
  });
});
