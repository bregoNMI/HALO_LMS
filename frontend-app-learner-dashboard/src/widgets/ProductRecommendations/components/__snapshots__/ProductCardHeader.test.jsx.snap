// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRecommendations ProductCardHeader matches snapshot 1`] = `
<div>
  <Hyperlink
    className="base-card-link text-decoration-none"
    destination="http://localhost:18000/executive-education?linked_from=recommender"
    onClick={[Function]}
  >
    <div
      className="d-flex align-items-center border-bottom"
    >
      <h3
        className="h3 mb-2 text-left"
      >
        Executive Education
      </h3>
      <Icon
        className="text-primary-500 ml-2.5"
        src={[MockFunction icons.ChevronRight]}
      />
    </div>
  </Hyperlink>
  <p
    className="text-gray-500 x-small mt-2 mb-2"
  >
    Short Courses to develop leadership skills
  </p>
</div>
`;
